common/models/user.js

```diff
@@ -669,6 +669,42 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    UserModel.observe('before save', function beforeEmailUpdate(ctx, next) {
+      var oldEmail, newEmail, oldId, newId;
+      if (ctx.isNewInstance) return next();
+
+      if (ctx.where) {
+        ctx.hookState.id = ctx.where.id;
+        oldId = ctx.hookState.id;
+      }
+
+      ctx.Model.find({ where: { id: oldId }}, function(err, userInstances) {
+        ctx.hookState.originalUserData = userInstances.map(function(u) {
+          return { id: u.id, email: u.email };
+        });
+      });
+      next();
+
+      UserModel.observe('after save', function afterEmailUpdate(ctx, next) {
```

Comment:
Actually, I think it's better to expire the tokens _before_ performing the email change. That way, if deleting access tokens fail, the email is not changed either.

@rmg @gunjpan any opinions on this?